### PR TITLE
refactor: switching to use ID Formatters in (some) Requires Import steps

### PR DESCRIPTION
**Pull request by tombuildsstuff**
This PR switches to using the ID Formatter value in the Requires Import step to ensure that this is consistent (and can be used in errors etc)

This is fairly large, but each service package is scoped to a commit, so this may be worth a commit-by-commit review.

Even with this open there's another ~400 total to fix, which'll come in subsequent PR's:

```
$ grep -R "\*existing.ID)" ./internal/services | wc -l
288

$ grep -R "d.SetId(\*read.ID)" ./internal/services | wc -l
96
```

### CODE REVIEW

File: internal/services/synapse/synapse_sql_pool_resource.go [hunk 1]
```diff
@@ -337,6 +331,7 @@ func resourceSynapseSqlPoolRead(d *pluginsdk.ResourceData, meta interface{}) err
 			d.SetId("")
 			return nil
 		}
+
 		return fmt.Errorf("retrieving Synapse SqlPool %q (Workspace %q / Resource Group %q): %+v", id.Name, id.WorkspaceName, id.ResourceGroup, err)
```
Comment (stephybun) [inline internal/services/synapse/synapse_sql_pool_resource.go:335]: Looks like this was one, the log message in [line 330](https://github.com/hashicorp/terraform-provider-azurerm/blob/b8312c1e64ef47d428814430d40d80d93aec71f4/internal/services/synapse/synapse_sql_pool_resource.go#L330) and [line 340](https://github.com/hashicorp/terraform-provider-azurerm/blob/b8312c1e64ef47d428814430d40d80d93aec71f4/internal/services/synapse/synapse_sql_pool_resource.go#L340) was forgotten
```
return fmt.Errorf("retrieving %s: %+v", id, err)
```

File: internal/services/servicebus/servicebus_namespace_disaster_recovery_config_resource.go [hunk 2]
```diff
@@ -163,7 +153,7 @@ func resourceServiceBusNamespaceDisasterRecoveryConfigUpdate(d *pluginsdk.Resour
 			return fmt.Errorf("issuing break pairing request for Service Bus Namespace Disaster Recovery Configs %q (Namespace %q / Resource Group %q): %s", id.DisasterRecoveryConfigName, id.NamespaceName, id.ResourceGroup, err)
```
Comment (stephybun) [inline internal/services/servicebus/servicebus_namespace_disaster_recovery_config_resource.go:153]: Forgot this one

File: internal/services/servicebus/servicebus_namespace_disaster_recovery_config_resource.go [hunk 3]
```diff
@@ -178,7 +168,7 @@ func resourceServiceBusNamespaceDisasterRecoveryConfigUpdate(d *pluginsdk.Resour
 		return fmt.Errorf("creating/updating Service Bus Namespace Disaster Recovery Configs %q (Namespace %q / Resource Group %q): %s", id.DisasterRecoveryConfigName, id.NamespaceName, id.ResourceGroup, err)
```
Comment (stephybun) [inline internal/services/servicebus/servicebus_namespace_disaster_recovery_config_resource.go:168]: and this one

File: internal/services/servicebus/servicebus_namespace_disaster_recovery_config_resource.go [hunk 4]
```diff
@@ -243,7 +233,7 @@ func resourceServiceBusNamespaceDisasterRecoveryConfigDelete(d *pluginsdk.Resour
 		return fmt.Errorf("breaking pairing for Service Bus Namespace Disaster Recovery Configs %q (Namespace %q / Resource Group %q): %s", id.DisasterRecoveryConfigName, id.NamespaceName, id.ResourceGroup, err)
```
Comment (stephybun) [inline internal/services/servicebus/servicebus_namespace_disaster_recovery_config_resource.go:233]: and this one

File: internal/services/netapp/netapp_volume_resource.go [hunk 5]
```diff
@@ -452,7 +447,7 @@ func resourceNetAppVolumeCreateUpdate(d *pluginsdk.ResourceData, meta interface{
 
 		// Wait for volume replication authorization to complete
 		log.Printf("[DEBUG] Waiting for replication authorization on NetApp Volume Provisioning Service %q (Resource Group %q) to complete", id.Name, id.ResourceGroup)
```
Comment (stephybun) [inline internal/services/netapp/netapp_volume_resource.go:449]: `log.Printf("[DEBUG] Waiting for replication authorization on NetApp Volume Provisioning Service %s", id)`

File: internal/services/netapp/netapp_volume_resource.go [hunk 6]
```diff
@@ -561,7 +556,7 @@ func resourceNetAppVolumeDelete(d *pluginsdk.ResourceData, meta interface{}) err
 
 			// Waiting for replication be in broken state
 			log.Printf("[DEBUG] Waiting for replication on NetApp Volume Provisioning Service %q (Resource Group %q) to be in broken state", replVolumeID.Name, replVolumeID.ResourceGroup)
```
Comment (stephybun) [inline internal/services/netapp/netapp_volume_resource.go:558]: `log.Printf("[DEBUG] Waiting for replication on NetApp Volume Provisioning Service %s to be in broken state", replVolumeID)`

File: internal/services/netapp/netapp_volume_resource.go [hunk 7]
```diff
@@ -572,7 +567,7 @@ func resourceNetAppVolumeDelete(d *pluginsdk.ResourceData, meta interface{}) err
 		}
 
 		log.Printf("[DEBUG] Waiting for replication on NetApp Volume Provisioning Service %q (Resource Group %q) to be deleted", replVolumeID.Name, replVolumeID.ResourceGroup)
```
Comment (stephybun) [inline internal/services/netapp/netapp_volume_resource.go:569]: This one also seems to be forgotten

File: internal/services/netapp/netapp_pool_resource.go [hunk 8]
```diff
@@ -196,29 +186,33 @@ func resourceNetAppPoolDelete(d *pluginsdk.ResourceData, meta interface{}) error
 	// In this case we're going to re-check status code again.
 	// For more details, see related Bug: https://github.com/Azure/azure-sdk-for-go/issues/6374
 	log.Printf("[DEBUG] Waiting for NetApp Pool %q (Resource Group %q) to be deleted", id.Name, id.ResourceGroup)
```
Comment (stephybun) [inline internal/services/netapp/netapp_pool_resource.go:188]: `log.Printf("[DEBUG] Waiting for NetApp Pool %s to be deleted", *id)`

**Review by stephybun — APPROVED** [review]
Thanks @tombuildsstuff. Flagged several log and error messages that seem to have been forgotten and the linter is unhappy about some `fmt.Errorf` calls. Other than that LGTM!

**Review by katbyte — APPROVED** [review]
LGTM 🏗️

**Pull request merged**